atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/memory/InMemoryAtlasDbConfig.java

```diff
@@ -15,22 +15,27 @@
  */
 package com.palantir.atlasdb.memory;
 
-import org.immutables.value.Value;
-
 import com.fasterxml.jackson.annotation.JsonTypeName;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.auto.service.AutoService;
 import com.palantir.atlasdb.spi.KeyValueServiceConfig;
 
-@JsonDeserialize(as = ImmutableInMemoryAtlasDbConfig.class)
 @JsonTypeName(InMemoryAtlasDbConfig.TYPE)
 @AutoService(KeyValueServiceConfig.class)
-@Value.Immutable
-public abstract class InMemoryAtlasDbConfig implements KeyValueServiceConfig {
+public final class InMemoryAtlasDbConfig implements KeyValueServiceConfig {
     public static final String TYPE = "memory";
 
     @Override
     public String type() {
         return TYPE;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        return this == other || (other != null && this.getClass() == other.getClass());
```

Comment:
The type field is static though, so will definitely be the same for other InMemoryAtlasDbConfigs (and to distinguish this from other types of config, the `getClass()` comparison will take care of that).

I'll add a test that checks that this is not equal to some other AtlasDbConfig.